arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Bracket.kt

```diff
@@ -32,7 +32,7 @@ public sealed class ExitCase {
  * @see guaranteeCase for registering a handler that executes for any [ExitCase].
  */
 public suspend inline fun <A> onCancel(
-  fa: suspend () -> A,
+  fa: () -> A,
```

Comment:
I think this is still needed.. Or what I've used to hit in the passed was a compiler bug ... 😅 Perhaps I should've suppressed with a comment at the time.

IIRC this is needed for method references, we can potentially also add a test so we can not accidentally break this in the future unless this was fixed in the compiler. For the old cases of:

```kotlin
onCancel(::awaitCancellation) { ... } 
```

This doesn't compile without `fa: suspend () -> A`.

Reply:
It definitely works now (at least in the examples and tests). The main choice for this removal was the compiler telling me that `suspend` was not super-great in an `inline` function. But we can bring it back if you feel this is breaking the way the library should be used.